plugins/@grouparoo/pardot/src/lib/client.ts

```diff
@@ -0,0 +1,138 @@
+import axios, { AxiosInstance } from "axios";
+import { escapeXML } from "./utils";
+
+interface PardotClientOptions {
+  businessUnitId: string;
+  loginUrl?: string;
+  pardotUrl?: string;
+}
+
+interface SalesforceUserInfo {
+  organizationName: string;
+  userFullName: string;
+}
+
+interface PardotCustomField {
+  id: number;
+  name: string;
+  fieldId: string;
+  type: string;
+  isRecordMultipleResponses: boolean;
+}
+
+class PardotClient {
+  private loginUrl: string;
+  private pardotUrl: string;
+  private businessUnitId: string;
+
+  userInfo?: SalesforceUserInfo;
+  request?: AxiosInstance;
+
+  constructor(options?: PardotClientOptions) {
+    this.loginUrl = options?.loginUrl || "https://login.salesforce.com";
+    this.pardotUrl = options?.pardotUrl || "https://pi.pardot.com";
+    this.businessUnitId = options.businessUnitId;
+  }
+
+  // Salesforce SOAP Login
+  async login(username: string, password: string, securityToken?: string) {
```

Comment:
I'm not against using `jsforce` if it makes things easier.

Reply:
I don't have a strong opinion here. My reasoning here was basically since we're not using `jsforce` for anything else and this is what it does internally then we don't _need_ the dependency. Dealing with SOAP is kind of different than everything else though.

If you'd like me to replace this with `jsforce`, I'm not opposed to it.

Reply:
I figure it will likely to be updated more than we would know to do here. Use the same version as the salesforce plugin so they stay together.